### PR TITLE
[Enhancement] Improve various German translations

### PR DESCRIPTION
**Pull request by b-fein**
### Motivation and Context
While using Artemis in German I noticed various typos and additional spaces in compound-nouns. When I found inconsistencies (e.g. ‘Multiple Choice‘/‘Multiple-Choice’, ‘Drag And Drop Exercise’/‘Drag-And-Drop Exercise’, ‘Buildplan’/‘Build-Plan’) I chose one format.

This is most probably not an exhaustive fix, so:
If you find any more fixable translations, feel free to add your commits to this PR or add a comment below.

### Description
In general, I tried to apply these rules when fixing compound-nouns:
* If one part is German the other English (and also not common as loanword), I combined them using a dash. E.g. ‘Build-Plan’, ‘Studierenden-Repository’.
* If all parts are German and they can be combined into a proper noun, I did so without dashes. E.g. ‘Kurs Verwaltung‘ → ‘Kursverwaltung’, ‘Bewertung Kriterium’ → ‘Bewertungskriterium’.
* If this resulted in a weird combination, I used a mix. E.g. ‘Beispiel Lösungs Editor’ → ’Beispiellösungs-Editor’.

### Steps for Testing
Set the language to German. Click through the various pages in Artemis to check if a translation looks out-of-place or if you can spot additional ones that should be changed.

### Test Coverage
n/a

### Screenshots
n/a

### CODE REVIEW

File: src/main/webapp/i18n/de/assessmentInstructions.json [hunk 1]
```diff
@@ -4,15 +4,15 @@
             "instructions": "Instruktionen",
             "editor": {
                 "addCredits": "[credits] Punkte",
-                "addInstruction": "[instruction] Bewertung Instruktionen",
+                "addInstruction": "[instruction] Bewertungsinstruktionen",
```
Comment (SimonEntholzer) [inline src/main/webapp/i18n/de/assessmentInstructions.json:7]: Here we could maybe use `Bewertungsinstruktion` as it is singular?

File: src/main/webapp/i18n/de/featureOverview.json [hunk 2]
```diff
@@ -1,9 +1,9 @@
 {
     "featureOverview": {
-        "title": "Online Klausuren in Artemis",
-        "featureDetails": "Feature Details",
+        "title": "Onlineklausuren in Artemis",
+        "featureDetails": "Feature-Details",
         "instructor": {
-            "pageTitle": "Klausur Features für Instructors",
+            "pageTitle": "Klausurfeatures für Instructors",
```
Comment (SimonEntholzer) [inline src/main/webapp/i18n/de/featureOverview.json:6]: I guess we could use this:
```suggestion
            "pageTitle": "Klausurfeatures für Instruktoren",
```

**Review by SimonEntholzer — APPROVED** [review]
Wow that's a lot of really good changes! I never realized there are so many mistakes.
I have two small comments, but overall I approve :+1:

**Comment by krusche** [comment]
Thanks, we really appreciate your effort here.

**Pull request merged**